### PR TITLE
tpm2: add internal key sealer interface

### PR DESCRIPTION
**Pull request by chrisccoulson**
This is a new internal interface that is going to be used by the new key
sealing APIs in order to increase code use between them.

### CODE REVIEW

File: tpm2/key_sealer.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"github.com/canonical/go-tpm2"
+	"github.com/canonical/go-tpm2/templates"
+	"github.com/canonical/go-tpm2/util"
+
+	"golang.org/x/xerrors"
+)
+
+// keySealer is an abstraction for creating a sealed key object
+type keySealer interface {
+	CreateSealedObject(data []byte, nameAlg tpm2.HashAlgorithmId, policy tpm2.Digest) (tpm2.Private, *tpm2.Public, tpm2.EncryptedSecret, error)
```
Comment (pedronis) [inline tpm2/key_sealer.go:32]: this probably merits a comment so that the argument can be described a bit

File: tpm2/key_sealer.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	if err != nil {
+		return nil, nil, nil, xerrors.Errorf("cannot create sealed object: %w", err)
+	}
+	return priv, pub, nil, err
+}
+
+// importableObjectKeySealer is an implementation of keySealer that seals keys to
+// a storage key in the form of an importable object.
+type importableObjectKeySealer struct {
+	tpmKey *tpm2.Public
```
Comment (pedronis) [inline tpm2/key_sealer.go:83]: this merits a comment, I notice that SealKeyToExternalTPMStorageKey had quite a long comment

**Review by pedronis — COMMENTED** [review]
thanks, some remarks about having a few more comments

**Comment by chrisccoulson** [comment]
I've added some extra comments to this now.

**Pull request merged**